### PR TITLE
[TACHYON-709] Fix deadlock because different blocks may share the same lock on worker side

### PR DESCRIPTION
**Pull request by shimingfei**
https://tachyon.atlassian.net/browse/TACHYON-709

### CODE REVIEW

File: servers/src/main/java/tachyon/worker/block/BlockMetadataManagerView.java [hunk 1]
```diff
@@ -91,14 +94,19 @@ public boolean isBlockPinned(long blockId) {
    * @return boolean, true if block is locked
    */
   public boolean isBlockLocked(long blockId) {
-    return mLockedBlocks.contains(blockId);
+    int index = BlockLockManager.blockHashIndex(blockId);
+    if (index <= mInUseLocks.length()) {
```
Comment (calvinjia) [inline servers/src/main/java/tachyon/worker/block/BlockMetadataManagerView.java]: Should this be `<`?

File: servers/src/main/java/tachyon/worker/block/BlockStoreMeta.java [hunk 2]
```diff
@@ -83,7 +83,7 @@ public long getCapacityBytes() {
 
   public int getNumberOfBlocks() {
     int numberOfBlocks = 0;
-    for (List blockIds : mBlockIdsOnDirs.values()) {
+    for (List<Long> blockIds : mBlockIdsOnDirs.values()) {
```
Comment (calvinjia) [inline servers/src/main/java/tachyon/worker/block/BlockStoreMeta.java]: Is this just a style fix?

File: servers/src/main/java/tachyon/worker/block/BlockMetadataManagerView.java (hunk 1 above)
Comment (shimingfei) [inline servers/src/main/java/tachyon/worker/block/BlockMetadataManagerView.java]: yes, it should be <

File: servers/src/main/java/tachyon/worker/block/BlockStoreMeta.java (hunk 2 above)
Comment (shimingfei) [inline servers/src/main/java/tachyon/worker/block/BlockStoreMeta.java]: yes, it is a style fix

**Comment by calvinjia** [comment]
+1 LGTM, thanks for the fix!

**Comment by apc999** [comment]
+1 LGTM!  thanks for the patch

**Pull request merged**